src/core/analytics.js

```diff
@@ -868,15 +907,16 @@ class Analytics {
         );
       }
 
+      const clonedRudderElement = R.clone(rudderElement);
       // convert integrations object to server identified names, kind of hack now!
-      transformToServerNames(rudderElement.message.integrations);
-      rudderElement.message.integrations = getMergedClientSuppliedIntegrations(
-        this.integrationsData,
-        clientSuppliedIntegrations,
-      );
+      transformToServerNames(clonedRudderElement.message.integrations);
 
-      // self analytics process, send to rudder
-      this.eventRepository.enqueue(rudderElement, type);
+      // Holding the cloud mode events based on flag and integrations load check
+      if (!this.bufferDataPlaneEventsUntilReady || this.clientIntegrationObjects) {
```

Comment:
does this.clientIntegrationObjects implies that integrations are loaded?

Reply:
yes, it's an indicator which we are using for device mode events as well

Reply:
better to create new variable denoting the lolig instead of placing the logiic inside confitionals.

Something like:
const shouldNotBufferDataPlaneEvents = !this.bufferDataPlaneEventsUntilReady || this.clientIntegrationObjects